### PR TITLE
entgql: move the order-fields logic to template function

### PR DESCRIPTION
**Pull request by a8m**
If the target `Edge.Type` does not have order fields, they won't be defined in the GraphQL schema, and having them in the method signature will break the gqlgen binding.

### CODE REVIEW

File: entgql/template.go [hunk 1]
```diff
@@ -261,3 +264,14 @@ func removeOldTemplate(g *gen.Graph, name string) error {
 	}
 	return nil
 }
+
+// annotation extracts the entgql.Annotation or returns its empty value.
+func annotation(ants gen.Annotations) (*Annotation, error) {
```
Comment (giautm) [inline entgql/template.go:269]: please use `decodeAnnotation` function,

Comment (a8m) [reply entgql/template.go:269]: decode feels to me like an internal detail. From the caller's perspective, I'm interested to get the `entgql.Annotation` and don't care how it's done.

Comment (giautm) [reply entgql/template.go:269]: That function is existes in the codebase, so re-use it or rename it, please.

Comment (a8m) [reply entgql/template.go:269]: oh, got it. I didn't see the `decodeAnnotation` function. Thanks

File: entgql/annotation.go [hunk 2]
```diff
@@ -173,15 +173,13 @@ func (a *Annotation) Decode(annotation interface{}) error {
...
+// annotation extracts the entgql.Annotation or returns its empty value.
+func annotation(ants gen.Annotations) (*Annotation, error) {
 	ant := &Annotation{}
-	if annotations == nil || annotations[ant.Name()] == nil {
-		return ant, nil
-	}
-
-	if err := ant.Decode(annotations[ant.Name()]); err != nil {
-		return nil, err
+	if ants != nil || ants[ant.Name()] != nil {
```
Comment (giautm) [inline entgql/annotation.go:179]:
```suggestion
	if ants != nil && ants[ant.Name()] != nil {
```

**Review by giautm — CHANGES_REQUESTED** [review]
I think there is a mistake

File: entgql/annotation.go (hunk 2 above)
Comment (a8m) [reply entgql/annotation.go]: Oops. Thanks @giautm

**Review by giautm — APPROVED** [review]
LGTM, <3

**Pull request merged**